nixos/modules/services/monitoring/vmalert.nix

```diff
@@ -0,0 +1,124 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.vmalert;
+  mkConfigFile =
+    pkgs.writeText "rules.yaml" (lib.generators.toYAML { } cfg.configuration);
+
+  checkedConfig = file:
+    pkgs.runCommand "checked-config" { buildInputs = [ cfg.package ]; } ''
+      ln -s ${file} $out
+    '';
+
+  vmalertRulesYAML = let
+    yml = if cfg.configText != null then
+      pkgs.writeText "rules.yaml" cfg.configText
+    else
+      mkConfigFile;
+  in checkedConfig yml;
+
+in {
+
+  options.services.vmalert = with lib; {
+
+    enable = mkEnableOption "vmalert";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.victoriametrics;
+      defaultText = "pkgs.victoriametrics";
+      description = ''
+        The VictoriaMetrics distribution to use.
+      '';
+    };
+
+    notifierUrls = mkOption {
+      type = types.listOf types.str;
+      default = [ "http://localhost:9093" ];
+      description = ''
+        AlertManager URL
+      '';
+    };
+
+    remoteWriteUrl = mkOption {
+      type = types.str;
+      default = "http://localhost:8428";
+      description = ''
+        remote write compatible storage to persist rules
+      '';
+    };
+
+    remoteReadUrl = mkOption {
+      type = types.str;
+      default = "http://localhost:8428";
+      description = ''
+        PromQL compatible datasource to restore alerts state from
+      '';
+    };
+
+    dataSourceUrl = mkOption {
+      type = types.str;
+      default = "http://localhost:8428";
+      description = ''
+        PromQL compatible datasource
+      '';
+    };
+
+    configuration = mkOption {
+      type = types.nullOr types.attrs;
+      default = null;
+      description = ''
+        vmalert configuration as nix attribute set.
+      '';
+    };
+
+    configText = mkOption {
```

Comment:
Is this needed? Is there any configuration that can't be created by the `configuration` option? Do we need this? Is it redundant?